### PR TITLE
Rename Application model to ServiceProvider

### PR DESCRIPTION
**Pull request by pkarman**
Also zaps lots of unused i18n strings.

**Why**:

Naming consistency across apps

### CODE REVIEW

File: app/views/user_mailer/user_new_service_provider.html.slim [hunk 1]
```diff
@@ -3,10 +3,10 @@ html
     meta content='text/html; charset=UTF-8' http-equiv='Content-Type'
   body
 
-  p You have created a new Identity service provider application:
+  p You have created a new Identity service provider service_provider:
```
Comment (amoose) [inline app/views/user_mailer/user_new_service_provider.html.slim]: Too many service providers!

File: config/routes.rb [hunk 2]
```diff
@@ -15,7 +15,7 @@
   end
 
   namespace :users do
-    resources :applications do
+    resources :service_providers do
     end
   end
```
Comment (amoose) [inline config/routes.rb]: I don't think we need to address this any time soon, but note that service provider entries will need to have delegated access at some point.

**Comment by amoose** [comment]
Nice work. 👌

File: config/routes.rb (hunk 2 above)
Comment (pkarman) [inline config/routes.rb]: good point. My intention was to eventually add `Group` or `Org` in order to define authz for a group of `User`s.

File: app/views/user_mailer/user_new_service_provider.html.slim (hunk 1 above)
Comment (pkarman) [inline app/views/user_mailer/user_new_service_provider.html.slim]: nice catch. search+replace FAIL. fixed now.

**Comment by pkarman** [comment]
@amoose thanks for the review. Think I've hit all your comments.

**Pull request merged**